Bluetooth: Controller: Fix Sync Failed to be Established on no memory

cvinayak opened the pull request:
Update implementation to generate Periodic Sync Failed to be
Established when Sync Established message could not be
generate due to lack of free node rx buffers and when there
is sync lost before sync established message could be
generated.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

cvinayak committed:
Bluetooth: Controller: Fix Sync Failed to be Established on no memory

Update implementation to generate Periodic Sync Failed to be
Established when Sync Established message could not be
generate due to lack of free node rx buffers and when there
is sync lost before sync established message could be
generated.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

ppryga commented on subsys/bluetooth/controller/ll_sw/ull_sync_types.h:
```diff
@@ -21,6 +21,8 @@ struct ll_sync_set {
 	uint16_t volatile timeout_reload; /* Non-zero when sync established */
 	uint16_t timeout_expire;
 
+	void (*lll_sync_prepare)(void *param);
```
```suggestion
	/* Member to store periodic advertising sync prepare.
	 * Also serves as a flag to inform if sync established was
	 * already generated.
	 */
	void (*lll_sync_prepare)(void *param);
```

ppryga commented on subsys/bluetooth/controller/ll_sw/ull_sync.c:
```diff
@@ -1235,9 +1233,17 @@ static void ticker_stop_sync_lost_op_cb(uint32_t status, void *param)
+	struct ll_sync_set *sync;
 	struct node_rx_pdu *rx;
 
+	/* sync established was not generated yet, no free node rx */
```
Is this the only one condition that the `sync->lll_sync_prepare` may differ from `lll_sync_prepare`, I mean no sync established event due to lack of free `node_rx`?

cvinayak replied:
yes, `sync->lll_sync_prepare` is set to `lll_sync_prepare` when ULL generates the sync established event towards HCI here https://github.com/zephyrproject-rtos/zephyr/pull/42572/files#diff-05ca44e3fbcc4f5bd0abd6e72153924632b01ba62979cf73b118d92cc3c836c7R833

cvinayak committed:
Bluetooth: Controller: Add sync established code comment

Add code comment related to sync context field used to
indicate the sync established event being generated towards
HCI Layer.

Co-authored-by: Piotr Pryga <piotr.pryga@nordicsemi.no>
Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>
